tests/unit/s2n_connection_serialize_test.c

```diff
@@ -473,6 +500,488 @@ int main(int argc, char **argv)
+            client_conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(client_conn);
+            EXPECT_SUCCESS(s2n_connection_deserialize(client_conn, buffer, sizeof(buffer)));
+
+            /* Lets do a resumption handshake with the acquired ticket. Wipe both connections
+             * so that new handshake can happen. */
+            EXPECT_SUCCESS(s2n_connection_wipe(server_conn));
+            EXPECT_SUCCESS(s2n_connection_wipe(client_conn));
+            EXPECT_SUCCESS(s2n_connection_set_config(client_conn, resumption_config));
+            EXPECT_SUCCESS(s2n_connection_set_config(server_conn, resumption_config));
+
+            /* Client sets up a resumption connection with the received session ticket data */
+            EXPECT_SUCCESS(s2n_connection_set_session(client_conn, session_ticket.data, session_ticket.size));
+
+            /* Free and re-initialize io pair. We don't reuse the io_pair to properly mimic
+             * sending a serialized connection off-box */
+            EXPECT_OK(s2n_io_stuffer_pair_free(&io_pair));
+            EXPECT_OK(s2n_io_stuffer_pair_init(&io_pair));
+            EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, server_conn, &io_pair));
+
+            /* Negotiate successful resumption connection */
+            EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
+            EXPECT_FALSE(IS_FULL_HANDSHAKE(client_conn));
+            EXPECT_FALSE(IS_FULL_HANDSHAKE(server_conn));
+
+            EXPECT_OK(s2n_io_stuffer_pair_free(&io_pair));
+            EXPECT_SUCCESS(s2n_connection_free(client_conn));
+        };
+
+        /* Server is serialized. Can write a session ticket after deserialization and resumption can
+         * occur with it. */
+        {
+            DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(client_conn);
+            struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(server_conn);
+
+            EXPECT_SUCCESS(s2n_connection_set_config(client_conn, resumption_config));
+            EXPECT_SUCCESS(s2n_connection_set_config(server_conn, resumption_config));
+
+            struct s2n_test_io_stuffer_pair io_pair = { 0 };
+            EXPECT_OK(s2n_io_stuffer_pair_init(&io_pair));
+            EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, server_conn, &io_pair));
+
+            EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
+            EXPECT_EQUAL(s2n_connection_get_actual_protocol_version(server_conn), S2N_TLS13);
+
+            /* Get the client to do a read to pick up the first session ticket. We won't use this one;
+             * we're interested in the tickets the server sends after the transfer. */
+            EXPECT_OK(s2n_send_and_recv_test(server_conn, client_conn));
```

Comment:
Should the resumption config maybe not be set on the original server, so that an original session ticket isn't sent? This might help ensure that a ticket was definitely sent by the deserialized server and used to resume.